handler/transport/backend.go

```diff
@@ -239,12 +240,40 @@ func (b *Backend) evalTransport(req *http.Request) *Config {
 		}
 	}
 
-	originURL, _ := url.Parse(origin)
+	originURL, err := url.Parse(origin)
+	if err != nil {
+		b.upstreamLog.LogEntry().WithField("hcl", "backend").Error(err)
+	}
+
+	if rawURL, ok := req.Context().Value(request.URLAttribute).(string); ok {
+		urlAttr, err := url.Parse(rawURL)
+		if err != nil {
+			b.upstreamLog.LogEntry().WithField("hcl", "backend").Error(err)
+		}
+
+		if origin != "" && urlAttr.Scheme+"://"+urlAttr.Host != origin {
+			b.upstreamLog.LogEntry().WithField("hcl", "backend").Error(
+				"The host of 'url' and 'backend.origin' must be equal",
+			)
+
+			return nil, fmt.Errorf("The host of 'url' and 'backend.origin' must be equal")
+		}
+
+		originURL.Host = urlAttr.Host
+		originURL.Scheme = urlAttr.Scheme
+		req.URL.Scheme = urlAttr.Scheme
+		req.URL.Path = urlAttr.Path // TODO: evalURLPath()???
```

Comment:
No eval required, this is done at producer/request before setting the url context